data/src/main/java/io/gnosis/data/backend/dto/TransactionDto.kt

```diff
@@ -1,16 +1,60 @@
 package io.gnosis.data.backend.dto
 
+import pm.gnosis.model.Solidity
+
 data class DataDecodedDto(
     val method: String,
-    val parameters: List<ParamsDto>?
+    val parameters: List<ParamDto>?
 )
 
-data class ParamsDto(
-    val name: String,
-    val type: String,
-    val value: Any
+data class ValueDecodedDto(
+    val operation: Operation,
+    val to: Solidity.Address,
+    val value: Long,
+    val data: String,
+    val dataDecoded: DataDecodedDto?
 )
 
+sealed class ParamDto {
+    abstract val type: String
+    abstract val name: String
+    abstract val value: Any?
+
+    data class AddressParam(
+        override val type: String,
+        override val name: String,
+        override val value: Solidity.Address
+    ) : ParamDto()
+
+    data class ArrayParam(
+        override val type: String,
+        override val name: String,
+        override val value: List<Any>
+    ) : ParamDto()
+
+    data class BytesParam(
```

Comment:
Not sure I see the point for `BytesParam` if you already have a `ValueParam` that is a flat scalar value.

Reply:
BytesParam is for "bytes" type and it can have valueDecoded field. ValueParam only has value field for a flat scalar value